### PR TITLE
numpy instance __setstate__

### LINKED ISSUE #18141: __setstate__

### PR DESCRIPTION
**Pull request by KartikeyDubeyKD**
closes #18141

### CODE REVIEW

**Comment by KartikeyDubeyKD** [comment]
#18141

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 1]
```diff
@@ -2969,6 +2969,34 @@ def test_numpy_instance_setitem(
     )
 
 
+#__setstate__
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="numpy.array",
+    method_name="__setstate__",
+    state={"ivy_array": np.array},
```
Comment (Daniel4078) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2977]: you should use helper functions to generate inputs for arguments, not listing it out directly like this. you can refer to other tests in the file for examples.
_ test_numpy_instance_setstate[cpu-ivy.functional.backends.tensorflow-False-False] _
ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2975: in wrapped_test
    init_tree="numpy.array",
E   hypothesis.errors.InvalidArgument: test_numpy_instance_setstate() got an unexpected keyword argument 'state', from `state={'ivy_array': <built-in function array>}` in @given

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello! I found something that you need to fix for the tests to pass. Please check out the comments below.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 1 above)
Comment (KartikeyDubeyKD) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2977]: Ok got it, sir, 
Thanks for the review.

Comment (KartikeyDubeyKD) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2977]: @sladyn98

Comment (Daniel4078) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2977]: I just found out that maybe your implementation of _setstate_ may not be correct as it is different form the definition of numpy version(https://numpy.org/doc/stable/reference/generated/numpy.ndarray.__setstate__.html)

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Sorry, but it seems that the Ivy bot just unassign me for some reason and get you someone who are not available. But after some research, I found that your implemenation needs to be changed in the first place.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 1 above)
Comment (KartikeyDubeyKD) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2977]: sorry for the delay response, I'm working on it

**Comment by hmahmood24** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**